rfcs/2024-08-21-motion-presets.md

```diff
@@ -0,0 +1,161 @@
+| **Hardware Accelarated Animations**      | ✅ (Hybrid - WAAPI for some transformations with fallback to JS) [Hardware Accelarated POC](#hardware-accelarated-motion-using-framer-motion) | ✅ (Built on WAAPI)                           | ❌                                     | ✅                                            |
+| **Easy to implement complex animations** | ✅ (Declarative API)                                                                                                                          | ❌                                            | ❌                                     | ❌                                            |
+| **React Router Page Transition**         | ✅                                                                                                                                            | ❌ (No native support but can be implemented) | ✅                                     | ❌ (No native support but can be implemented) |
+| **Morph Animations / Layout Animations** | ✅ [Framer Motion POC](#morph-animations-with-framer-motion)                                                                                  | ❌                                            | ✅                                     | ❌                                            |
+| **Small bundle size**                    | ❌ (4.6kb core + 40kb features that can be lazy loaded)                                                                                       | ✅ (4kb )                                     | ❌ (26kb core + features)              | ✅ (0kb)                                      |
+
+There is also detailed comparison of these libraries at [Motion One Docs - Feature Comparisons](https://motion.dev/docs/feature-comparison#comparison-table)
+
+## POCs
+
+### Hardware Accelarated Motion using Framer Motion
+
+https://github.com/user-attachments/assets/5aa7ba3f-666e-449c-ae1d-821f93c12962
+
+<p align="right">
+  <a href="https://codesandbox.io/p/sandbox/framer-motion-enter-animation-forked-rk3tvd?file=%2Fsrc%2FApp.tsx%3A27%2C28"><img src="https://codesandbox.io/static/img/play-codesandbox.svg" /></a>
+</p>
+
+### Hardware Accelarated Motion using GSAP
+
+> [!NOTE]
+>
+> While GSAP does offload some work to hardware using CSS, it still requires javascript to work and stops working if JS thread is blocked
+
+https://github.com/user-attachments/assets/57593b19-3a19-4713-b768-9bcbf69097ee
+
+<p align="right">
+<a href="https://codesandbox.io/p/sandbox/framer-motion-hardware-accelarated-animations-forked-7qjyg9">
+  <img alt="Edit GSAP: Hardware Accelarated Animations " src="https://codesandbox.io/static/img/play-codesandbox.svg">
+</a>
+</p>
+
+### Morph / Layout Animations with Framer Motion
+
+https://github.com/user-attachments/assets/ad3d4a23-c3b9-4980-a051-a0f44e7224dc
```

Comment:
Try doing the morph animation with blade components?

Reply:
Yup will do that POC with API decision now

Reply:
We should try to replicate the Switch animation and see how our primitives plays out. 

Switch has a 2 stage animation

Click Hold (Elongate the circle) -> Release (De-elongate the circle + move to the opposite side) 

Scale, transformX both will have to animate at once.